version-upgrade-operator/README.md

```diff
@@ -0,0 +1,673 @@
+  observedGeneration: 1
+
+  conditions:
+    # Ready condition: Indicates if the controller is able to reconcile this
+    # resource. False might mean invalid spec or other fundamental issues preventing reconciliation attempts.
+    - type: Ready
+      status: "True" # "True", "False", "Unknown"
+      observedGeneration: 1
+      lastTransitionTime: "2025-04-15T08:30:00Z"
+      reason: "ReconciliationSucceeded" # CamelCase reason code, must present also errors related to referenced resources: like Schedule or versionSource missing or errored
+      message: "VersionUpgradeConfig is operational" # Human-readable summary
+
+    # Synced condition: Indicates if the target resources are currently at the
+    # latest discovered version according to the configuration. False could mean an
+    # upgrade is pending, failed, or paused by the schedule/policy.
+    - type: Synced
+      status: "False" # "True", "False", "Unknown"
+      observedGeneration: 1
+      lastTransitionTime: "2025-04-15T09:05:00Z"
+      reason: "SyncPausedBySchedule" # TargetsSynced, TargetsSyncing, SyncPausedBySchedule, SyncPausedByNotBefore, SyncFailed
+      message: "Upgrade paused, current time outside the scheduled window defined by utcSchedule."
+
+  # lastReconciliationTime records when the controller last attempted to process
+  # this resource. Useful for debugging timing issues.
+  lastReconciliationTime: "2025-04-15T09:10:15Z"
+
+  # versionSource provides details about the last check against the version source.
+  versionSource:
+    # lastCheckTime records when the version source (e.g., OCIRepository) was last checked.
+    lastCheckTime: "2025-04-15T09:10:10Z"
+    # latestDiscoveredVersion holds the most recent valid SemVer found in the source.
+    # This might be different from the version currently applied to targets.
+    latestDiscoveredVersion: "0.45.1"
+
+  # scheduleSource provides specific timestamps when action will be taken based on the linked VUS object
+  scheduleSource:
+    # nextUpgradeWindowStart provides the calculated start time of the *next*
+    # window during which upgrades are permitted according to the spec.
+    nextUpgradeWindowStartTime: "2025-04-16T04:00:00Z" # Calculated from schedule
+    # nextUpgradeWindowEnd provides the calculated end time of that next window.
+    # (Typically start time + duration)
+    nextUpgradeWindowEndTime: "2025-04-16T04:05:00Z" # Calculated from startTime + duration
+    # lastCheckTime records when the schedule source was last checked.
+    lastCheckTime: "2025-04-15T09:10:10Z"
+
+  # TODO: think if the status is not too verbose, some info is redundant
+
+  # upgradeAllowed indicates if, based on the suspend, current time and valid the controller
+  # is currently permitted to perform an upgrade if one is needed.
+  upgradeAllowed: false
+
+  # lastAppliedVersion shows what was the version tried during the last version-applying sync
+  lastAppliedVersion: "0.45.0"
```

Comment:
Maybe we can use this to some of the problems above, but it feels a little risky relying on the status of the object, becasue this is not something recoverable, but something that was and can only be added at special reconcilation loops - not all of them - that takes place during the upgrade window.

If the VUC is deleted then put back by Flux lets say, this info is gone forever.